### PR TITLE
feat: list objects & buckets

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

support list objects by id and list buckets by id

### Rationale

N/A

### Example

```
curl --location --request POST 'localhost:9133/?objects-query' \
--header 'Authorization: authTypeV2 ECDSA-secp256k1, Signature=1234567812345678123456781234567812345678123456781234567812345678' \
--header 'Content-Type: application/json' \
--data-raw '{
    "ids": [333,666]
}'
```

```
curl --location --request POST 'localhost:9133/?buckets-query' \
--header 'Authorization: authTypeV2 ECDSA-secp256k1, Signature=1234567812345678123456781234567812345678123456781234567812345678' \
--header 'Content-Type: application/json' \
--data-raw '{
    "ids": [222]
}'
```

### Changes

Notable changes: 
* support list objects by id and list buckets by id

### CODE REVIEW

File: base/gfspclient/metadata.go [hunk 1]
```diff
@@ -404,3 +404,49 @@ func (s *GfSpClient) GetGroupList(
 	}
 	return resp.Groups, resp.Count, nil
 }
+
+func (s *GfSpClient) ListBucketsByBucketID(
+	ctx context.Context,
+	bucketIDs []int64,
+	includeRemoved bool,
+	opts ...grpc.DialOption) ([]*types.Bucket, error) {
```
Comment (sysvm) [inline base/gfspclient/metadata.go:412]: These function params can be moved to one line, don't exceed 120 columns in one line.

File: base/gfspclient/metadata.go [hunk 2]
```diff
@@ -404,3 +404,49 @@ func (s *GfSpClient) GetGroupList(
...
+		return nil, ErrRpcUnknown
+	}
+	return resp.Buckets, nil
+}
+
+func (s *GfSpClient) ListObjectsByObjectID(
+	ctx context.Context,
+	objectIDs []int64,
+	includeRemoved bool,
+	opts ...grpc.DialOption) ([]*types.Object, error) {
```
Comment (sysvm) [inline base/gfspclient/metadata.go:435]: ditto

File: modular/gater/metadata_handler.go [hunk 3]
```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
 	w.Header().Set(ContentTypeHeader, ContentTypeJSONHeaderValue)
 	w.Write(b.Bytes())
 }
+
+// listObjectsByObjectIDHandler list objects by object ids
+func (g *GateModular) listObjectsByObjectIDHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err       error
+		b         bytes.Buffer
```
Comment (sysvm) [inline modular/gater/metadata_handler.go:514]: rename b to buf is better.

File: modular/gater/metadata_handler.go [hunk 4]
```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
...
+	}
+
+	objects, err = g.baseApp.GfSpClient().ListObjectsByObjectID(reqCtx.Context(), objectIDs.IDs, false)
+	if err != nil {
+		log.Errorf("failed to list objects by ids", "error", err)
+		return
+	}
+
+	grpcResponse := &types.GfSpListObjectsByObjectIDResponse{Objects: objects}
+
```
Comment (sysvm) [inline modular/gater/metadata_handler.go:551]: meaningless blank line.

File: modular/gater/metadata_handler.go [hunk 5]
```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
...
+	}
+
+	buckets, err = g.baseApp.GfSpClient().ListBucketsByBucketID(reqCtx.Context(), bucketIDs.IDs, false)
+	if err != nil {
+		log.Errorf("failed to list buckets by ids", "error", err)
+		return
+	}
+
+	grpcResponse := &types.GfSpListBucketsByBucketIDResponse{Buckets: buckets}
+
```
Comment (sysvm) [inline modular/gater/metadata_handler.go:603]: ditto

Comment (BarryTong65) [reply modular/gater/metadata_handler.go:602]: fixed

File: modular/gater/metadata_handler.go (hunk 4 above)
Comment (BarryTong65) [reply modular/gater/metadata_handler.go:551]: fixed

File: modular/gater/metadata_handler.go (hunk 3 above)
Comment (BarryTong65) [reply modular/gater/metadata_handler.go]: fixed

File: base/gfspclient/metadata.go (hunk 2 above)
Comment (BarryTong65) [reply base/gfspclient/metadata.go]: fixed

File: base/gfspclient/metadata.go (hunk 1 above)
Comment (BarryTong65) [reply base/gfspclient/metadata.go]: fixed

**Review by will-2012 — APPROVED** [review]

**Review by annielz — APPROVED** [review]

**Pull request merged**